src/dune_cache_storage/layout.ml

```diff
@@ -82,7 +82,10 @@ let value_storage_dir = Versioned.value_storage_dir Version.Value.current
 let value_path = Versioned.value_path Version.Value.current
 
 let create_cache_directories () =
-  List.iter
-    [ temp_dir; metadata_storage_dir; file_storage_dir; value_storage_dir ]
-    ~f:(fun path -> ignore (Fpath.mkdir_p (Path.to_string path) : Fpath.mkdir_p_result))
+  [ temp_dir; metadata_storage_dir; file_storage_dir; value_storage_dir ]
+  |> Result.List.iter ~f:(fun path ->
+    match Fpath.mkdir_p (Path.to_string path) with
+    | Already_exists | Created -> Ok ()
+    | (exception Unix.Unix_error (EACCES, _, _))
+    | (exception Unix.Unix_error (EROFS, _, _)) -> Error path)
```

Comment:
I looked at the man page for `mkdir` and the amount of error codes possible is massive. What about just catching every unix error? All of them can have the same fallback.

Reply:
Ok, I'm passing the error to the caller so the caller can display the error reason (and the path that failed).